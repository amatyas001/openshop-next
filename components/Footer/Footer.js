import {
  Flex,
  SimpleGrid,
  Box,
  Heading,
  Text,
  List,
  ListItem,
  ListIcon,
} from '@chakra-ui/core';
import { FaHandsHelping, FaAt, FaShoppingBag } from 'react-icons/fa';

export const Footer = () => {
  return (
    <>
      <Flex
        minHeight='200px'
        borderTop='1px'
        borderBottom='1px'
        bordercolor='gray.100'
        color='gray.200'
        bg='black'
        py='2rem'
        px='5rem'
        justifyContent='space-around'
        flexWrap='wrap'
      >
        <Box minWidth='200px'>
          <Heading fontSize='1.5rem'>
            <Text as={FaHandsHelping} mb='-5px' mr='5px' />
            support
          </Heading>
          <List spacing={3}>
            <ListItem>
              <ListIcon color='gray.500' />
              Terms of Service
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Privacy Policy
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Sitemap
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Helpdesk
            </ListItem>
          </List>
        </Box>
        <Box minWidth='200px'>
          <Heading fontSize='1.5rem'>
            <Text as={FaAt} mb='-5px' mr='5px' />
            social
          </Heading>
          <List spacing={3}>
            <ListItem>
              <ListIcon color='gray.500' />
              Facebook
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Twitter
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Pinterest
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Youtube
            </ListItem>
          </List>
        </Box>
        <Box minWidth='200px'>
          <Heading fontSize='1.5rem'>
            <Text as={FaShoppingBag} mb='-5px' mr='5px' />
            openshop
          </Heading>
          <List spacing={3}>
            <ListItem>
              <ListIcon color='gray.500' />
              24/7 Service
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              About the company
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Contact us
            </ListItem>
            <ListItem>
              <ListIcon color='gray.500' />
              Report an issue
            </ListItem>
          </List>
        </Box>
      </Flex>
      <Flex bg='black' height='70px'>
        <Text d='block' m='auto' color='gray.200'>
          This is a sample webiste made by <strong>Mátyás Angyal</strong>
          {' - '}
          <strong>amatyas001@gmail.com</strong>
        </Text>
        <Text m='auto' color='gray.200'>
          <strong>
            <em>React / Next / Node / Git / Netlify </em>
          </strong>
        </Text>
      </Flex>
    </>
  );
};
