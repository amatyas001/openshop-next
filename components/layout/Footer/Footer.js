import Link from 'next/link';
import {
  Flex,
  Box,
  Heading,
  Link as ChakraLink,
  List,
  ListItem,
  SimpleGrid,
} from '@chakra-ui/core';
import { FOOTER_SECTIONS } from '@app/config';
import * as COLORS from '@app/config/colors';

/**
 * Displays a footer layout element at the bottom every page
 *
 * @see https://amatyas001.github.io/openshop-next/#section-configuration
 */
export const Footer = (props) => {
  return (
    <>
      <SimpleGrid
        as='footer'
        data-testid='footer'
        minHeight='200px'
        py='2rem'
        px={{ sm: '2%' }}
        bg={COLORS.BG.dark}
        color={COLORS.TEXT.dark}
        borderTop='1px'
        borderTopColor={COLORS.SPACER.dark}
        columns={FOOTER_SECTIONS.length}
        {...props}
      >
        {FOOTER_SECTIONS.map((section) => (
          <Flex
            data-testid='footer-section'
            as='section'
            key={section.title}
            mx='auto'
            flexDirection='column'
          >
            <Heading data-testid='footer-section-title' fontSize='1.5rem'>
              <Box
                data-testid='footer-section-icon'
                as={section.icon}
                mb='-5px'
                mr='5px'
              />
              {section.title}
            </Heading>
            <List spacing={3} px={{ sm: '0' }} mb={{ sm: '50px', lg: '20px' }}>
              {section.elements.map((element) => (
                <ListItem data-testid='footer-element' key={element.text}>
                  <Link href={element.link} passHref>
                    <ChakraLink
                      data-testid='footer-element-link'
                      color={COLORS.TEXT.dark}
                    >
                      {element.text}
                    </ChakraLink>
                  </Link>
                </ListItem>
              ))}
            </List>
          </Flex>
        ))}
      </SimpleGrid>
    </>
  );
};
