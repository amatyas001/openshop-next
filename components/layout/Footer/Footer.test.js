import { create, act } from 'react-test-renderer';
import { Footer } from './Footer';

describe('<Footer />', () => {
  let tree;
  beforeAll(() => {
    act(() => {
      tree = create(<Footer />);
    });
  });

  it('should render without props', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
