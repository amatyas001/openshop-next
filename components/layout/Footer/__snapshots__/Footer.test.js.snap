// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render without props 1`] = `
<footer
  className="css-1wmngoz"
  data-testid="footer"
>
  <section
    className="css-1705j0v"
    data-testid="footer-section"
  >
    <h2
      className="css-1xnp38i"
      data-testid="footer-section-title"
    >
      <svg
        className="css-2b1hth"
        data-testid="footer-section-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M488 192H336v56c0 39.7-32.3 72-72 72s-72-32.3-72-72V126.4l-64.9 39C107.8 176.9 96 197.8 96 220.2v47.3l-80 46.2C.7 322.5-4.6 342.1 4.3 357.4l80 138.6c8.8 15.3 28.4 20.5 43.7 11.7L231.4 448H368c35.3 0 64-28.7 64-64h16c17.7 0 32-14.3 32-32v-64h8c13.3 0 24-10.7 24-24v-48c0-13.3-10.7-24-24-24zm147.7-37.4L555.7 16C546.9.7 527.3-4.5 512 4.3L408.6 64H306.4c-12 0-23.7 3.4-33.9 9.7L239 94.6c-9.4 5.8-15 16.1-15 27.1V248c0 22.1 17.9 40 40 40s40-17.9 40-40v-88h184c30.9 0 56 25.1 56 56v28.5l80-46.2c15.3-8.9 20.5-28.4 11.7-43.7z"
        />
      </svg>
      support
    </h2>
    <ul
      className="css-18u2w7e"
    >
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Terms of Service
        </a>
      </li>
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Privacy Policy
        </a>
      </li>
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Sitemap
        </a>
      </li>
      <li
        className="css-992tcj"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Helpdesk
        </a>
      </li>
    </ul>
  </section>
  <section
    className="css-1705j0v"
    data-testid="footer-section"
  >
    <h2
      className="css-1xnp38i"
      data-testid="footer-section-title"
    >
      <svg
        className="css-2b1hth"
        data-testid="footer-section-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C118.941 8 8 118.919 8 256c0 137.059 110.919 248 248 248 48.154 0 95.342-14.14 135.408-40.223 12.005-7.815 14.625-24.288 5.552-35.372l-10.177-12.433c-7.671-9.371-21.179-11.667-31.373-5.129C325.92 429.757 291.314 440 256 440c-101.458 0-184-82.542-184-184S154.542 72 256 72c100.139 0 184 57.619 184 160 0 38.786-21.093 79.742-58.17 83.693-17.349-.454-16.91-12.857-13.476-30.024l23.433-121.11C394.653 149.75 383.308 136 368.225 136h-44.981a13.518 13.518 0 0 0-13.432 11.993l-.01.092c-14.697-17.901-40.448-21.775-59.971-21.775-74.58 0-137.831 62.234-137.831 151.46 0 65.303 36.785 105.87 96 105.87 26.984 0 57.369-15.637 74.991-38.333 9.522 34.104 40.613 34.103 70.71 34.103C462.609 379.41 504 307.798 504 232 504 95.653 394.023 8 256 8zm-21.68 304.43c-22.249 0-36.07-15.623-36.07-40.771 0-44.993 30.779-72.729 58.63-72.729 22.292 0 35.601 15.241 35.601 40.77 0 45.061-33.875 72.73-58.161 72.73z"
        />
      </svg>
      social
    </h2>
    <ul
      className="css-18u2w7e"
    >
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Faceboook
        </a>
      </li>
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Twitter
        </a>
      </li>
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Pinterest
        </a>
      </li>
      <li
        className="css-992tcj"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Youtube
        </a>
      </li>
    </ul>
  </section>
  <section
    className="css-1705j0v"
    data-testid="footer-section"
  >
    <h2
      className="css-1xnp38i"
      data-testid="footer-section-title"
    >
      <svg
        className="css-2b1hth"
        data-testid="footer-section-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
        />
      </svg>
      openshop
    </h2>
    <ul
      className="css-18u2w7e"
    >
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          24/7 Service
        </a>
      </li>
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          About Openshop
        </a>
      </li>
      <li
        className="css-1mips0e"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Contact Us
        </a>
      </li>
      <li
        className="css-992tcj"
        data-testid="footer-element"
      >
        <a
          className="css-1wwlnhi"
          data-testid="footer-element-link"
          href="#0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Report Issue
        </a>
      </li>
    </ul>
  </section>
</footer>
`;
